core/src/transaction.rs

```diff
@@ -74,141 +66,158 @@ impl rlp::Decodable for UnverifiedTransaction {
     }
 }
 
+impl SignedTransaction {
+    /// Append object with a signature into RLP stream
+    fn rlp_append_sealed_transaction(&self, s: &mut RlpStream) {
+        s.begin_list(6);
+        s.append(&self.unsigned.seq);
+        s.append(&self.unsigned.fee);
+        s.append(&self.unsigned.network_id);
+        s.append(&self.unsigned.action);
+        s.append(&self.sig);
+        s.append(&self.signer_public);
+    }
+}
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct VerifiedTransaction(SignedTransaction);
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct UnverifiedTransaction(SignedTransaction);
+
+impl rlp::Encodable for VerifiedTransaction {
+    fn rlp_append(&self, s: &mut RlpStream) {
+        self.0.rlp_append(s);
+    }
+}
+
+impl rlp::Decodable for VerifiedTransaction {
+    fn decode(d: &Rlp<'_>) -> Result<Self, DecoderError> {
+        let transaction = d.as_val()?;
+        let verified = UnverifiedTransaction(transaction).try_into().expect("Invalid signature");
+        Ok(verified)
+    }
+}
+
+impl From<VerifiedTransaction> for UnverifiedTransaction {
+    fn from(tx: VerifiedTransaction) -> Self {
+        UnverifiedTransaction(tx.0)
+    }
+}
+
+impl VerifiedTransaction {
+    /// Signs the transaction as coming from `signer`.
+    pub fn new_with_sign(tx: Transaction, private: &Private) -> VerifiedTransaction {
+        let sig = sign(&tx.hash(), private);
+        UnverifiedTransaction::new(tx, sig, private.public_key())
+            .try_into()
+            .expect("secret is valid so it's recoverable")
```

Comment:
Let's modify the comment because we're not using the previous key recovery

Reply:
I modified this comment to `The transaction's signature is invalid". How about this?